pkg/controller/dns/external_dns_reconciler.go

```diff
@@ -20,19 +25,71 @@ func newExternalDnsReconciler(manager ctrl.Manager, resources []client.Object) e
 
 // NewExternalDns starts all resources required for external dns
 func NewExternalDns(manager ctrl.Manager, conf *config.Config, self *appsv1.Deployment) error {
-	if conf.DNSZoneDomain == "" {
+	if len(conf.DNSZoneIDs) == 0 {
 		return nil
 	}
 
-	dnsConfig := &manifests.ExternalDnsConfig{
-		ResourceName:  "external-dns",
-		TenantId:      conf.TenantID,
-		Subscription:  conf.DNSZoneSub,
-		ResourceGroup: conf.DNSZoneRG,
-		Domain:        conf.DNSZoneDomain,
-		RecordId:      conf.DNSRecordID,
-		IsPrivate:     conf.DNSZonePrivate,
-	}
-	objs := manifests.ExternalDnsResources(conf, self, dnsConfig)
+	configs := parseZoneIds(conf)
```

Comment:
parse the zone ids higher up. We should do this in the config object.

We will use the zones in other places so it makes sense to move the parsing up

Reply:
do you mean load info about which zones are private/public and then the resource groups and subscriptions directly into the config object?